internal/datastore/mysql/datastore.go

```diff
@@ -118,7 +118,29 @@ func newMySQLDatastore(ctx context.Context, uri string, options ...Option) (*Dat
 		return nil, errors.New("error in NewMySQLDatastore: connection URI for MySQL datastore must include `parseTime=true` as a query parameter; see https://spicedb.dev/d/parse-time-mysql for more details")
 	}
 
-	connector, err := mysql.MySQLDriver{}.OpenConnector(uri)
+	// Setup the credentials provider
+	var credentialsProvider datastore.CredentialsProvider
+	if config.credentialsProviderName != "" {
+		credentialsProvider, err = datastore.NewCredentialsProvider(ctx, config.credentialsProviderName)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if credentialsProvider != nil {
+		// we must transmit the token over the connection, and not a hash
+		parsedURI.AllowCleartextPasswords = true
```

Comment:
newline after this line.

Also, will this be the case for all credentials providers or just this one? We should log a warning here too

Reply:
Good point. Say there is a new credentials provider that pulls the "normal" password from something like Vault, then this would not need to be set.

It looks like this can be set in the connection string as well using `allowCleartextPasswords=true`, so another option is to remove this and require the user to set it when they use AWS IAM auth. If they miss it it's not obvious though since the server just rejects it with access denied - had to pull out tcpdump to catch this one.

Reply:
If it will always apply to the AWS creds provider, I say we return another param from the call that indicates whether the token is plaintext (or not) and switch based on that

Reply:
Sound good. I opted to add a flag on the `CredentialsProvider` since we need to know this ahead of retrieving the actual token.